components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3527,6 +3525,40 @@ void RewardsServiceImpl::OnGetAllMonthlyReportIds(
   std::move(callback).Run(ids);
 }
 
+void RewardsServiceImpl::GetAllContributions(
+    GetAllContributionsCallback callback) {
+  bat_ledger_->GetAllContributions(
+      base::BindOnce(&RewardsServiceImpl::OnGetAllContributions,
+                     AsWeakPtr(),
+                     std::move(callback)));
+}
+
+void RewardsServiceImpl::OnGetAllContributions(
+    GetAllContributionsCallback callback,
+    ledger::ContributionInfoList contributions) {
+  std::vector<brave_rewards::ContributionInfo> converted;
+  for (const auto& contribution_item : contributions) {
+    brave_rewards::ContributionInfo contribution;
+    contribution.contribution_id_ = contribution_item->contribution_id;
+    contribution.amount_ = contribution_item->amount;
+    contribution.type_ = static_cast<int>(contribution_item->type);
+    contribution.step_ = static_cast<int>(contribution_item->step);
+    contribution.retry_count_ = contribution_item->retry_count;
+    contribution.created_at_ = contribution_item->created_at;
+    contribution.processor_ = static_cast<int>(contribution_item->processor);
+    for (const auto& publisher_item : contribution_item->publishers) {
+      brave_rewards::ContributionPublisher publisher;
+      publisher.contribution_id_ = publisher_item->contribution_id;
+      publisher.publisher_key_ = publisher_item->publisher_key;
+      publisher.total_amount_ = publisher_item->total_amount;
+      publisher.contributed_amount_ = publisher_item->contributed_amount;
+      contribution.publishers_.push_back(std::move(publisher));
+    }
+    converted.push_back(contribution);
```

Comment:
Should this be a move instead of a copy?

Reply:
Yes, fixed.